### PR TITLE
fix(gatsby-source-drupal): don't show "file download" activity if there are no files to download

### PR DESCRIPTION
**Pull request by skippednote**
If the nodes don't have any files attached like images Gatsby would throw the following error
```
"gatsby-source-drupal" threw an error while running the sourceNodes lifecycle:
Parameter `array` must have at least one item
```

To prevent false warnings we are checking if there are any images to download to begin with.

### CODE REVIEW

**Comment by skippednote** [comment]
Fixes #15761

**Comment by pieh** [comment]
Hey @skippednote! 

Similar pull request was just merged ( https://github.com/gatsbyjs/gatsby/pull/16160 ), but I think we could still merge this in with small change/improvement. Right now `downloadingFilesActivity.start()` and `downloadingFilesActivity.end()` are outside of the `if (fileNodes.length)` conditional block - if you could move those inside, we could merge this PR as well!

File: packages/gatsby-source-drupal/src/gatsby-node.js [hunk 1]
```diff
@@ -140,16 +140,15 @@ exports.sourceNodes = async (
   downloadingFilesActivity.start()
```
Comment (pieh) [inline packages/gatsby-source-drupal/src/gatsby-node.js]: Please move this inside `if (fileNodes.length)` conditional code path

File: packages/gatsby-source-drupal/src/gatsby-node.js [hunk 2]
```diff
@@ -140,16 +140,15 @@ exports.sourceNodes = async (
...
       await downloadFile(
         { node, store, cache, createNode, createNodeId },
         pluginOptions
       )
-    }
-  )
+    })
+  }
 
   downloadingFilesActivity.end()
```
Comment (pieh) [inline packages/gatsby-source-drupal/src/gatsby-node.js]: Please move this inside `if (fileNodes.length)` conditional code path

Comment (skippednote) [reply packages/gatsby-source-drupal/src/gatsby-node.js]: @pieh Thank you for catching that. I've made those changes.

**Pull request merged**